### ISSUE TITLE
Improve consistency of model API

### ISSUE THREAD

**Issue by nalimilan** [JuliaStats/StatsBase.jl#280]
Our current `StatisticalModel` API is heavily inspired from R, adapted to match Julia rules (plus some improvements), but it is not really consistent (since R isn't either). Below is a summary of the situation. Names are considered OK if they are either a standard abbreviation, or are spelled in full (and the name is not too long).

Names consistent with R and OK: `coef`, `nobs`, `deviance`, `confint`, `fitted`, `residuals`, `predict`, `aic`, `bic`, `aicc`, `r2`/`r²`, `adjr2`, `adjr²`, `stderr`

Names that do not exist in R and are OK: `aicc`, `r2`/`r²`, `adjr2`, `adjr²`, `stderr`, `nulldeviance`

Names different from R for good reasons: `dof`, `dof_residual` (to avoid two-letter objects, limiting confusion with `df = DataFrame(...)`)

Names not consistent with R for bad reasons, and proposed change:
- `loglikelihood` -> `loglik` (shorter, relatively standard abbreviation, and consistent with R)
- `nullloglikelihood` -> `null_loglik` (shorter and more readable; if changed, `nulldeviance` should be called `null_deviance`)

Names consistent with R which should be changed:
- `vcov` -> `cov`
- `model_response` -> `response` ("model" is redundant with multiple dispatch)

Name to add and which does not exist in R: `params`, `param` or `parameters`. The choice depends on whether we accept abbreviations, or prefer full names everywhere. Names which should be changed accordingly include potentially `coef`, `loglikelihood`/`nullloglikelihood`, `nobs`, `adjr2`/`adjr²`, `confint` and `dof`.

**Comment by rofinn** [comment]
I like all of this except that I'm not a fan of the `lik` convention in R (maybe `like`?). Also, seems a little weird to have `null_loglik` and `null_deviance` when everything else seems to skip the `_`. Maybe we could make `null` a keyword which defaults to `false`?

**Comment by rofinn** [comment]
This is probably a separate issue, but would it make sense to merge the `StatisticalModel` API into [StatsModels.jl](https://github.com/JuliaStats/StatsModels.jl/blob/master/REQUIRE)?

**Comment by nalimilan** [comment]
> I like all of this except that I'm not a fan of the lik convention in R (maybe like?). Also, seems a little weird to have null_loglik and null_deviance when everything else seems to skip the _. Maybe we could make null a keyword which defaults to false?

Yeah. I guess whether to abbreviate `loglikelihood`depends on what we decide to do for other abbreviations. Regarding the `_`, I suggested this mainly because `nullloglikelihood` is really weird, but OTOH it isn't used that frequently. The idea of replacing it with a parameter is certainly worth considering.


> This is probably a separate issue, but would it make sense to merge the StatisticalModel API into StatsModels.jl?

We've discussed this in the past, but I think it's still an open question. I think @kleinschmidt wanted to put only table-related functions in that package, which would mean modeling packages wouldn't even have to depend on it: it would just handle the transformation of formulas + data to a model matrix. Anyway, not really the place to discuss this.

**Comment by nalimilan** [comment]
See also https://github.com/JuliaStats/Roadmap.jl/issues/4.
